ansible-playbook-base/roles/common/tasks/setup_10g_nic.yaml

```diff
@@ -0,0 +1,41 @@
+---
+- hosts: servers
+  tasks:
+    - name: Configures the first 10G nic with an active link
+      shell: |
+        for dev in $(ls /sys/class/net/)
+        do
+          link=$(ethtool ${dev} | grep "Link detected: yes" &> /dev/null; echo $?)
+          if [ $link -eq 0 ]
+          then
+            speed=$(ethtool ${dev} | grep "Speed: 10000Mb/s" &> /dev/null; echo $?)
+            if [ $speed -eq 0 ]
+              then
+                export MY_NIC=${dev}
+                break
+            fi
+          fi
+        done
+        echo "DEVICE=${MY_NIC}" > /etc/sysconfig/network-scripts/ifcfg-${MY_NIC}
+        echo "BOOTPROTO=none" >> /etc/sysconfig/network-scripts/ifcfg-${MY_NIC}
+        echo "ONBOOT=yes" >> /etc/sysconfig/network-scripts/ifcfg-${MY_NIC}
+        echo "TYPE=Ethernet" >> /etc/sysconfig/network-scripts/ifcfg-${MY_NIC}
+        echo "PREFIX=24" >> /etc/sysconfig/network-scripts/ifcfg-${MY_NIC}
+        echo "IPADDR=${MY_IP}" >> /etc/sysconfig/network-scripts/ifcfg-${MY_NIC}
```

Comment:
MY_IP here would be replaced with the variable confipv4addr when defined in the setup-profile, like,
 https://github.com/gluster/gbench/blob/master/ansible-playbook-base/roles/common/tasks/ipv4oib.yml#L8